component/core/src/main/java/org/exoplatform/social/core/jpa/storage/RDBMSIdentityStorageImpl.java

```diff
@@ -705,14 +705,17 @@ public int getIdentitiesForMentionsCount(String providerId, ProfileFilter profil
 
   @Override
   public int getIdentitiesByProfileFilterCount(String providerId, ProfileFilter profileFilter) throws IdentityStorageException {
-    ExtendProfileFilter xFilter = new ExtendProfileFilter(profileFilter);
-    xFilter.setProviderId(providerId);
-    ListAccess<IdentityEntity> list = getIdentityDAO().findIdentities(xFilter);
-
-    try {
-      return list.getSize();
-    } catch (Exception e) {
-      return 0;
+    if (profileFilter.isExcludeExternal()) {
+      return getIdentityDAO().getAllIdsCountByProvider(providerId);
```

Comment:
I think it's the contrary, this case = `!profileFilter.isExcludeExternal()`

Reply:
**getIdentityDAO().getAllIdsCountByProvider** returns the count of internal users, so this case is valid.
Test case added !

Reply:
> returns the count of internal users

I think in this case, the method name isn't the good one. It's better to add new method for this count or rename the existing one if it's used only here.
IMO, you should use all time the same method from IdentityDAO to return the count.
In addition, no need to retrieve a `ListAccess` Object and then call `getSize` on it. This will make the code more complicated and very hard to maintain.

Reply:
The method has been renamed, this case is only used when opening the People page (no filter, no search). In the another cases I don't touch the actual process.
I think it's quite complicated to be modified,  So I prefer not to be touched in this PR.

Reply:
Updated !